docs/rationale.md

```diff
@@ -23,4 +23,69 @@ try (Scope scope = TracingContextUtils.currentContextWith(span)) {
 ```
 
 It would not be possible to call `recordException` if `span` was also using try-with-resources.
-Because this is a common usage for spans, we do not support try-with-resources.
\ No newline at end of file
+Because this is a common usage for spans, we do not support try-with-resources.
+
+
+## Versioning and Releases
+
+### Assumptions
+
+- This project uses semver v2, as does the rest of OpenTelemetry.
+
+### Goals
+
+- API Stability: 
+    - Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, that API module will 
+function, *with no recompilation required*, with any SDK that has the same major version, and equal or greater minor or patch version. 
+    - For example, libraries that are instrumented with `opentelemetry-api-trace:1.0.1` will function with 
+SDK library `opentelemetry-sdk-trace:1.11.33`. 
+    - We call this requirement the "ABI" compatibility requirement for "Application Binary Interface" compatibility. 
+- SDK Stability:
+    - Public portions of the SDK (constructors, configuration, und-user interfaces) must remain backwards compatible. 
```

Comment:
I think these first two points aren't unique to SDK

Reply:
I was thinking this is a lesser requirement than the API requirements.

Reply:
I probably need to update this to be more clear. I think rather than "public", we need to provide a way to specify what pieces of the SDK are under this backwards-compatibility requirement and what isn't. Unless we put the whole SDK into one package, I think it will be hard to make sure that all public classes/interfaces are fully supported.